lua/nvim-treesitter/indent.lua

```diff
@@ -36,15 +36,35 @@ local function get_indents(bufnr)
     end
   end
 
-  return indent_nodes, branch_nodes
+  return { indents = indent_nodes, branches = branch_nodes }
 end
 
+local get_indents = (function()
+  local cache = {}
```

Comment:
couldn't you use `memoize_by_buf_tick`?

Reply:
i think i can try again but last time it wasn't working at all.

Reply:
yep i'm not sure why, but it fails to indent the whole buffer, only the on_bytes callback is working

Reply:
i fixed the memoize_by_buf_tick function to take only the bufnr as first argument. Not sure if this is the expected behavior but it didn't work properly otherwise.